### DISCUSSION TITLE
Firebase .getInt() fails after Internet is reconnected - ESP8266

### DISCUSSION THREAD

**Discussion by heisen-bug** [community/community#22996] [New to GitHub]
<p>Hi. I’m working on a project where I need to read data from my Firebase database frequently. I use a NodeMCU (ESP8266-12) to connect to a WiFi network and read data from the Firebase. If the Internet is available initially, everything works perfect. I get proper data from Firebase. But if I disconnect the Internet (NOT WiFi), and reconnect it again, then Firebase.getInt() would still return 0, and not the value stored in my database. I checked Firebase.success() function and it also returns 0 after Internet is reconnected. But the odd thing is, Firebase.setInt() works. ie, data is written into Firebase properly even after reconnection. </p>
<p>This is the code:</p>
<p><span class="hashtag">#include</span> &lt;ESP8266WiFi.h&gt;<br>
<span class="hashtag">#include</span> &lt;FirebaseArduino.h&gt;</p>
<p>// Set these to run example.<br>
<span class="hashtag">#define</span> FIREBASE_HOST “mydatabasehost”<br>
<span class="hashtag">#define</span> FIREBASE_AUTH “authentication”<br>
<span class="hashtag">#define</span> WIFI_SSID “OnePlus 3”<br>
<span class="hashtag">#define</span> WIFI_PASSWORD “cristiano”</p>
<p>void setup() {</p>
<p>Serial.begin(9600);</p>
<p>// connect to wifi.<br>
WiFi.begin(WIFI_SSID, WIFI_PASSWORD);<br>
Serial.print(“connecting”);<br>
while (WiFi.status() != WL_CONNECTED) {<br>
Serial.print(".");<br>
delay(500);<br>
}<br>
Serial.println();<br>
Serial.print("connected: ");<br>
Serial.println(WiFi.localIP());</p>
<p>Firebase.begin(FIREBASE_HOST, FIREBASE_AUTH);</p>
<p>}</p>
<p>void loop() {</p>
<p>Serial.println(Firebase.getInt(“manual”));<br>
if(Firebase.failed())<br>
{<br>
Serial.println(“Internet lost”);<br>
delay(5000);<br>
}<br>
Firebase.setInt(“x”,20);<br>
}</p>

**Comment by mpboom** [comment, marked as answer]
<p>I wouldn’t know why this happens, but why don’t you just reconnect to the WiFi network after you’ve detected the internet is lost (= just calling setup() again)?</p>
<p><em>PS: Next time could you please put your code in a code block? Using the &lt;/&gt; icon in the toolbar?</em></p>

**Comment by heisen-bug** [comment]
<p>Thanks for the reply. I was thinking the same thing too, but in a different way. I thought about using an ESP.restart() once internet is reconnected. I tried it a couple of hours ago and looks like it worked. Restarting the whole system seemed to have fixed the issue. But still it’s weird that it fails to read from firebase even when internet is connected.</p>
<p>By the way, is there any way I could send an unsigned long data to firebase using Arduino library functions (Like Firebase.setInt()… I couldn’t find one for “long” ) ?</p>

**Comment by mpboom** [comment]
<p>For the first part, glad it worked fine!</p>
<p>For the second part of your new question: I really don’t know enough about Firebase to really comment on that, but a quick Google search turned up this: <a href="https://stackoverflow.com/questions/31930406/storing-long-type-in-firebase" rel="nofollow noopener">https://stackoverflow.com/questions/31930406/storing-long-type-in-firebase</a> - it might be of use. Otherwise I would recommend asking a new question on this forum, both so everything stays organized but also to make sure your new question gets the attention it deservers (because this one is already marked as solved, not many people will look at it anymore).</p>

**Comment by ThuNguyenvan278** [comment]
<p>I have the same problem. Also, sometime I’ve got Firebase .getInt() failed when power on. So I use ESP.restart() everytime when an Firebase.failed() </p>
<p>I don’t know why this happens but this trick works for me</p>

**Comment by gustavosinbandera1** [comment]
<p>you can use ArduinoJson and embedd the long var inside json object <a href="https://arduinojson.org/v5/example/" rel="nofollow noopener">https://arduinojson.org/v5/example/</a></p>

**Comment by huuthanhdtd** [comment]
<p>In my case, fix Firebase.cpp AND fix FirebaseArduino.cpp of firebasearduino libs and worked perfectly</p>
<ol>
<li>
<p>Change FirebaseArduino.cpp:</p>
<p>void FirebaseCall::analyzeError(char* method, int status, const std::string&amp; path_with_auth) {<br>
if (status != 200) {<br>
error_ = FirebaseError(status,<br>
std::string(method) + " " + path_with_auth +<br>
": " + http_-&gt;errorToString(status));<br>
}</p>
<p>}</p>
</li>
</ol>
<p>To:</p>
<pre><code>void FirebaseCall::analyzeError(char* method, int status, const std::string&amp; path_with_auth) {
if (status != 200) {
error_ = FirebaseError(status,
std::string(method) + " " + path_with_auth +
": " + http_-&gt;errorToString(status));
}
else {
error_ = FirebaseError();
}
}
</code></pre>
<ol start="2">
<li>
<p>Change FirebaseArduino.cpp:</p>
<p>int FirebaseArduino::getInt(const String&amp; path) {<br>
getRequest(path);<br>
if (failed()) {<br>
return 0;<br>
}<br>
return FirebaseObject(req_.get()-&gt;response().c_str()).getInt();<br>
}</p>
</li>
</ol>
<p>To:</p>
<pre><code>int FirebaseArduino::getInt(const String&amp; path) {
getRequest(path);
if (failed()) {
return -1;
}
return FirebaseObject(req_.get()-&gt;response().c_str()).getInt();
}
</code></pre>
<p>You can change all get function to: return -1 for your code</p>

**Comment by anirudhvasireddy** [comment]
<p>No need to reconnect the device to wifi. Just try to add some value to the firebase before checking the connection. This line will helps you.</p>
<p>void loop()</p>
<p>{</p>
<p>Firebase.setInt(“a”,0);</p>
<p>if(Firebase.failed())</p>
<p>{</p>
<p>…</p>
<p>}</p>
